cpp/lib/src/NGram.cc

```diff
@@ -316,9 +317,9 @@ void ClassifyLanguage(std::istream &input, std::vector<std::string> * const top_
 
     // Select the top scoring language and anything that's close (as defined by alternative_cutoff_factor):
     const double high_score(languages_and_scores[0].second);
-    top_languages->push_back(languages_and_scores[0].first);
-    for (unsigned i(1); i < languages_and_scores.size() and (languages_and_scores[i].second < alternative_cutoff_factor * high_score); ++i)
-        top_languages->push_back(languages_and_scores[i].first);
+    top_languages->emplace_back(languages_and_scores[0].first, languages_and_scores[0].second);
```

Comment:
Why not make top_languages be a vector of DetectedLanguage and add an operator< member function to it?

Reply:
not necessary